### ISSUE TITLE
Clarify what happens when VS Code detects a Screen Reader

### ISSUE THREAD

**Issue by swinster** [Microsoft/vscode#32166]
- VSCode Version: Code 1.14.2 (cb82febafda0c8c199b9201ad274e25d9a76874e, 2017-07-19T23:34:09.706Z)
- OS Version: Windows_NT ia32 10.0.16257
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
python|don|0.7.0
beautify|Hoo|1.1.1
rest-client|hum|0.15.1
prettify-json|moh|0.0.3
PowerShell|ms-|1.4.1
vscode-nginx|sha|0.3.1

---

Steps to Reproduce:

1.Open VSCOde
2.

Reproduces without extensions: Yes/No 
Yes

![image](https://user-images.githubusercontent.com/8142964/29101229-b40e3338-7ca8-11e7-9344-64ebc1ccfef8.png)

![image](https://user-images.githubusercontent.com/8142964/29101238-c58b244a-7ca8-11e7-954c-024b220a3f55.png)

![image](https://user-images.githubusercontent.com/8142964/29101299-31d15516-7ca9-11e7-9bfe-e5dc0c113755.png)

In addition (not sure if this is you of the PowerShell adding guys, but the automatic understanding of brackets has broken (i.e. when you where by a open bracket, vsCode would highlight eh relevant closed bracket.

**Comment by weinand** [comment]
@swinster the "Folder" in the title of your issue is confusing.
You probably mean "Folding" has stopped working, correct?

**Comment by weinand** [comment]
I cannot reproduce a folding problem in VS Code 1.14.2 and PowerShell 1.4.1 extension:

![2017-08-09_13-12-26](https://user-images.githubusercontent.com/1898161/29118972-8e6a3008-7d04-11e7-83ab-61837b856e1f.png)

**Comment by weinand** [comment]
@swinster is this issue always reproducible for you?

**Comment by swinster** [comment]
@weinand yes, every file I open.

I had a really odd issue the other day where the debugger kept throwing an error for code that no longer existed - which I posted here https://github.com/Microsoft/vscode/issues/32096 but you closed immediately. For some reason, this cleared up, although the bracket detection and folding are still failing. 

Form a  basic IT point of view, I uninstalled the the PS extension and re-install, but this made no difference. I haven't touched vscode though. 

P.S. I updated the title - somewhat dyslexic, sorry :(

**Comment by swinster** [comment]
I ma never 100% sure whether things are vscode's issue of the PS Extension issue, but please let me know if you would like to see anything else.

**Comment by swinster** [comment]
So, Python, XML, JSON ......

**Comment by swinster** [comment]
FWIW, here is the PS issue - https://github.com/PowerShell/vscode-powershell/issues/988

**Comment by swinster** [comment]
Installed the new version of VScode, 1.15, no joy
Uninstalled the 32 bit version of VSCode and install the 64 bit version of VSCode, still no joy. 

I guess this is something to do with a setting files

**Comment by aeschli** [comment]
Folding is indendation based, and not related to the powershell extension being present.
It looks like the folding setting is set to `false`.
Also check your workspace settings...

**Comment by swinster** [comment]
@aeschli did you see the screen capture of the `editor.folding` section above?

**Comment by swinster** [comment]
Or in deed in the work space settings, where there is nothing configured.

![image](https://user-images.githubusercontent.com/8142964/29184128-14e6bd56-7dfd-11e7-998b-dca75a9f5727.png)

**Comment by swinster** [comment]
Here is a JSON file
![image](https://user-images.githubusercontent.com/8142964/29184273-891a57fa-7dfd-11e7-80c7-f1ec68fdc433.png)

XML
![image](https://user-images.githubusercontent.com/8142964/29184298-97ee220c-7dfd-11e7-8dea-f86f691e739a.png)

and you have seen PS above

**Comment by aeschli** [comment]
Sorry, I can't reproduce and have not idea what this could be. I'm also not aware of any other users that reported the same.
- you said its reproducible with all extensions disabled
- did you try this with new installation of VSCode? You can also try the [insiders](https://code.visualstudio.com/insiders)

**Comment by swinster** [comment]
Argggghhhhh. ******* Windows. Reboot - it was as simple as that. 

No Idea how or why this was caused, but I had to reboot for something else and as soon as I started vscode  folding and bracket detection started working.

**Comment by aeschli** [comment]
If you see this again, please check the status bar if it says 'screenreader detected'.
I just learned that this would remove folding and bracket matching.

**Comment by swinster** [comment]
Well, well, well, well, well, well!!!

Lo and behold, yes a screen reader is detected and I have one installed (Read&Write Gold, for dyslexics (not visually impaired) - hence the odd title when I originally posted). Of course, rebooting meant this shut down so wasn't' active when vscode started. Indeed, when this issue occurs, Read&Write is running, even though the screen reader part is switched off. 

I did not expect that!

Thanks for helping out here. I think we can close this and add to the useful knowledge bucket.

**Comment by aeschli** [comment]
Assigning to @alexandrudima to see what can be done here. My guess we should show a more prominent dialog with information when the screen reader mode is turned on.

**Comment by adnansalehin** [comment]
Code folding was not working for me for ages now, and it was driving me crazy. So, I decided to do something about it today. After hours and hours of searching, I finally landed on this page. 

First of all, thank you for the solution, @aeschli 
There are two bugs involved which made the solution really difficult to find.
First of all, code folding doesn't work when a screen reader is detected. Secondly, a screen reader is detected when there are none. 

I solved the issue simply by turning accessibility support off. ("editor.accessibilitySupport": "off") 

This is an unintuitive solution for the code folding issue. I think a lot of people are missing out on the folding feature because of this. Please fix these two bugs.

I thank you in advance.

**Comment by alexandrudima** [comment]
We will now show a dialog:

![image](https://user-images.githubusercontent.com/5047891/30818860-0f1408c2-a21d-11e7-8160-c83527f8860e.png)
